src/main/java/bio/terra/service/snapshotbuilder/query/filtervariable/FunctionFilterVariable.java

```diff
@@ -43,22 +45,39 @@ public String renderSQL(CloudPlatformWrapper platform) {
   }
 
   public enum FunctionTemplate implements SqlExpression {
-    TEXT_EXACT_MATCH("CONTAINS_SUBSTR(<fieldVariable>, <value>)"),
+    TEXT_EXACT_MATCH(
+        "CONTAINS_SUBSTR(<fieldVariable>, <value>)", "CHARINDEX(<value>, <fieldVariable>) > 0"),
     // BigQuery fuzzy match pattern is "bqutil.fn.levenshtein(UPPER(<fieldVariable>),
     // UPPER(<value>))<5"
     TEXT_FUZZY_MATCH("dbo.Levenshtein(UPPER(<fieldVariable>), UPPER(<value>), 5)"),
     IN("<fieldVariable> IN (<value>)"),
     NOT_IN("<fieldVariable> NOT IN (<value>)");
 
-    private final String template;
+    private final String gcpTemplate;
+    private final String azureTemplate;
 
     FunctionTemplate(String template) {
-      this.template = template;
+      this(template, template);
+    }
+
+    FunctionTemplate(String gcpTemplate, String azureTemplate) {
+      this.gcpTemplate = gcpTemplate;
+      this.azureTemplate = azureTemplate;
     }
 
     @Override
     public String renderSQL(CloudPlatformWrapper platform) {
-      return template;
+      if (platform != null) {
```

Comment:
To stay away from nesting thoughts on this? 

```
public String renderSQL(CloudPlatformWrapper platform) {
    if (platform == null) {
        throw new InvalidRenderSqlParameter("SQL cannot be generated because the Cloud Platform is null.");
    }

    if (platform.isAzure()) {
        return azureTemplate;
    }

    if (platform.isGcp()) {
        return gcpTemplate;
    }

    throw new NotImplementedException("Cloud Platform not implemented.");
}

```

Reply:
Yeah I like this!